tools/Render1D.py

```diff
@@ -0,0 +1,255 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+#!/usr/bin/env python
+#Python script to render 1D data written using SpECTRE in format v2
+# This script originated from someone in Summer of 2018.  The format for volume
+# data files has changed and we can use as a model the GenerateXdmf.py in spectre/tools
+# as a model.  In addition we probably want to make this script compatible with both
+# h5py and the python wrappers of SpECTRE H5 functions.  To this end we may want to factor
+# even more functions out
+
+
+import glob
+import h5py
+import argparse
+import sys
+import os # Gives the script additional functionality but may be unnesecary
+import numpy as np
+import matplotlib as mpl
+mpl.use('Agg')
+import matplotlib.pyplot as plt
+import matplotlib.animation as animation
+try:
+    import Tkinter
+except ImportError:
+    import tkinter as Tkinter
+
+global line
+global title
+# May Want these to be arrays, may not matter but should see if performance will be
+# a big deal.
+time = []
+coords = []
+data = []
+
+
+def find_extrema_over_data_set(arr):
+
+    '''
+    Find max and min over a range of number arrays
+    :param arr: the array over which to find the max and min
+    This doesn't really work if there are nan valued data pts.
+    '''
+    # The last statment of the docstring may or may not be true
+
+    return np.nanmin(np.asarray(arr)), np.nanmax(np.asarray(arr))
+
+
+def init():
+
+    '''
+    Initialize the animation canvas
+    :return: empty line and title info
+    '''
+
+    line.set_data([], [])
+    title.set_text("")
+    return line, title
+
+
+def animate(iteration):
+    '''
+    Update the animation canvas
+    '''
+
+    title.set_text("t = %.5f" % time[iteration])
+    line.set_data(coords[iteration], data[iteration])
+    return line, title
+
+
+def parse_cmd_line():
+    '''
+    parse command-line arguments
+    :return: dictionary of the command-line args, dashes are underscores
+    '''
+
+    parser = argparse.ArgumentParser(description='Render 1-dimensional data',
+                                     formatter_class=
+                                     argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('--file-prefix', type=str, required=True,
+                        help='common prefix of all h5 files being used \
+                        in the case of a single h5 file, it is the file name \
+                        without the .h5 extension')
+    parser.add_argument('--var', type=str, required=True,
+                        help='name of variable to render. eg \'psi\', \
+                        \'psi-Error\' or \'psi-Analytic\'')
+    parser.add_argument('--time', type=int,required=False,
+                        help='if given renders the integer observation step \
+                        instead of an animation (optional)')
+    parser.add_argument('--save', type=str, required=False,
```

Comment:
The places this is used add the appropriate extension.  This behavior needs to be documented or removed.  (My preference would be to just write to the name given and depend on the user to give the correct extension.  This avoids files like foo.pdf.pdf that I tend to get from SpEC.)